app/util/cron.py

```diff
@@ -0,0 +1,58 @@
+import atexit
+import datetime
+import json
+import logging
+import sys
+
+from app import app
+from sqlalchemy import func
+from app.model.email import Email, EmailStatus
+from app.util.db import db
+from app.util.smtp import SMTP
+from apscheduler.schedulers.background import BackgroundScheduler
+from apscheduler.triggers.interval import IntervalTrigger
+
+logging.basicConfig()
+SEND_INTERVAL_SEC = 60
+
+
+def resend_mail():
+    with app.app_context():
+        since = datetime.datetime.utcnow() - datetime.timedelta(hours=1)
+        mails_failed = EmailStatus.query \
+            .filter(EmailStatus.delivered == False, EmailStatus.sent_at > since) \
+            .having(func.count_(EmailStatus.delivered) <= 2) \
```

Comment:
I think I understood idea but what's not clear to me is how querying the status API tells a user that the message is permanently failed and won't ever be retried?

Reply:
i rewrote algorithm for getting non-delivered messages. please check it out now